kernel/kernel-api/src/main/java/io/delta/kernel/client/ParquetHandler.java

```diff
@@ -94,4 +95,17 @@ CloseableIterator<DataFileStatus> writeParquetFiles(
             CloseableIterator<FilteredColumnarBatch> dataIter,
             long maxFileSize,
             List<Column> statsColumns) throws IOException;
+
+    /**
+     * Write the given data as a Parquet file. This call either succeeds in creating the file with
+     * given contents or no file is created at all. It won't leave behind a partially written file.
+     * <p>
+     * @param filePath  Fully qualified destination file path
+     * @param data      Iterator of {@link FilteredColumnarBatch}
+     * @throws FileAlreadyExistsException if the file already exists and {@code overwrite} is false.
```

Comment:
There isn't a requirement where we want to overwrite a Parquet file. It was kept off to avoid the implementation complexity. In future if needed we can always add a overload method.